### DISCUSSION TITLE
mixed receiver for type methods are problematic

### DISCUSSION THREAD

**Discussion by bluebrown** [ogen-go/ogen#686] [Q&A]
Hi, I noticed that ogen will use mixed receivers for the generated type methods. This is problematic, especially when using default response for errors.

If default response for errors is used, the handler interface will return a non pointer struct.

```go
type Handler interface {
	SerialsSerialGet(ctx context.Context, params SerialsSerialGetParams) (V1GetResponseHeaders, error)
}
```

This is problematic since this makes it impossible to do anything useful with the response in middleware.  For one, it is hard to check if the type satisfies some interface since the methods have mixed receivers and all the setters have pointer receiver. And second, even if you can find out that the response implements a given setter interface, you would mutate a copy of the response in the middleware. 

Hence, I think types should have generally pointer receiver for all methods and get passed around as pointer.

**Comment by ghost** [comment]
PR #696 should solve this problem.
